src/Facility.Core/Http/HttpClientService.cs

```diff
@@ -101,8 +103,66 @@ protected HttpClientService(HttpClientServiceSettings? settings, Uri? defaultBas
 			{
 				var contentType = mapping.RequestBodyContentType ?? requestHeaders?.GetContentType();
 				httpRequest.Content = GetHttpContentSerializer(requestBody.GetType()).CreateHttpContent(requestBody, contentType);
-				if (m_disableChunkedTransfer)
+				if (m_shouldCompressRequest(request))
```

Comment:
```suggestion
				if (m_shouldCompressRequest?.Invoke(request) ?? defaults.CompressRequests)
```
More efficient?

Reply:
Maybe? Is having `defaults.CompressRequests` being mutable OK? (I deliberately captured the value in the constructor, and it seemed "simpler" to have `m_shouldCompressRequest` non-null.)

Reply:
Or we can just get rid of the delegate.